File: Session13/Java/DFSofGraph.java
https://practice.geeksforgeeks.org/problems/depth-first-traversal-for-a-graph/1

class Solution {
    
    public void dfs(int node, boolean vis[], ArrayList<ArrayList<Integer>> adj, ArrayList<Integer> res) {
        res.add(node); 
        vis[node] = true; 
        for(Integer it: adj.get(node)) {
            if(vis[it] == false) {
                dfs(it, vis, adj, res); 
            }
        }
    }
    
    // Function to return a list containing the DFS traversal of the graph.
    public ArrayList<Integer> dfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj) {
        
        ArrayList<Integer> res = new ArrayList<>(); 
        boolean vis[] = new boolean[V]; 
        
        dfs(0, vis, adj, res); 
        
        return res;
    }
}

File: Session13/C++/DFSofGraph.cpp
https://practice.geeksforgeeks.org/problems/depth-first-traversal-for-a-graph/1

    void dfs(vector<int>& res, int node, vector<int>& visited, vector<int> adj[]) {
        visited[node] = true;
        res.push_back(node);
        for(int nbr:adj[node]) {
            if(!visited[nbr]) {
                dfs(res, nbr, visited, adj);
            }
        }
    }
    
  public:
    // Function to return a list containing the DFS traversal of the graph.
    vector<int> dfsOfGraph(int V, vector<int> adj[]) {

        vector<int> res;
        vector<int> visited(V, 0);
        
        dfs(res, 0, visited, adj);
        
        return res;
    }